api/v1/jobs/PKPJobController.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+/**
+ * @file api/v1/jobs/PKPJobController.php
+ *
+ * Copyright (c) 2014-2023 Simon Fraser University
+ * Copyright (c) 2003-2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPJobController
+ *
+ * @ingroup api_v1_jobs
+ *
+ * @brief Handle API requests for jobs
+ *
+ */
+
+namespace PKP\API\v1\jobs;
+
+use APP\facades\Repo;
+use PKP\security\Role;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use PKP\core\PKPBaseController;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Route;
+
+class PKPJobController extends PKPBaseController
+{
+    public function isSiteWide(): bool
+    {
+        return true;
+    }
+
+    public function getHandlerPath(): string
+    {
+        return 'jobs';
+    }
+
+    public function getRouteGroupMiddleware(): array
+    {
+        $roles = implode('|', [Role::ROLE_ID_SITE_ADMIN]);
+
+        return [
+            "has.user",
+            "has.roles:{$roles}",
+        ];
+    }
+
+    public function getGroupRoutes(): Void
+    {       
+        Route::get('all', $this->getJobs(...))
+            ->name('getJobs');
+        
+        Route::get('failed/all', $this->getFailedJobs(...))
+            ->name('getFailedJobs');
+        
+        Route::post('redispatch/all', $this->redispatchAllFailedJob(...))
+            ->name('redispatchAllFailedJob');
+
+        Route::post('redispatch/{jobId}', $this->redispatchFailedJob(...))
+            ->name('redispatchFailedJob');
+        
+        Route::delete('failed/delete/{jobId}', $this->deleteFailedJob(...))
+            ->name('deleteFailedJob');
+    }
+
+    /**
+     * Get all pending jobs in the queue waiting to get executed
+     * 
+     * @param   \Illuminate\Http\Request $request 
```

Comment:
The `@param and `@return` self-documentation doesn't add anything now that we have good typehints; please remove them (here and elsewhere).

Reply:
Removed where it seems unnecessary .